components/hab/src/command/pkg/binlink.rs

```diff
@@ -135,89 +145,165 @@ pub fn binlink_all_in_pkg(ui: &mut UI,
     Ok(())
 }
 
-fn is_dest_on_path<T: AsRef<Path>>(dest_dir: T) -> bool {
+fn is_dest_on_path(dest_dir: &Path) -> bool {
     if let Some(val) = env::var_os("PATH") {
-        env::split_paths(&val).any(|p| p == dest_dir.as_ref())
+        env::split_paths(&val).any(|p| p == dest_dir)
     } else {
         false
     }
 }
 
+#[allow(dead_code)]
 struct Binlink {
-    dest: PathBuf,
-    src:  PathBuf,
+    link_path:           PathBuf,
+    target_path:         PathBuf,
+    default_interpreter: PathBuf,
```

Comment:
💯 on the change from `dest`/`src`. I think this is *much* clearer. Do we need the `_path` suffix, though? The type already tells us we're dealing with a path, and it seems inconsistent with `default_interpreter`.